packages/flutter/lib/src/material/page_transitions_theme.dart

```diff
@@ -0,0 +1,367 @@
+/// transition animation.
+///
+/// Apps can configure the map of builders for [ThemeData.platformTheme]
+/// to customize the default [MaterialPageRoute] page transition animation
+/// for different platforms.
+///
+/// See also:
+///
+///  * [FadeUpwardsPageTransitionsBuilder], which defines a default page transition.
+///  * [OpenUpwardsPageTransitionsBuilder], which defines a page transition
+///    that's similar to the one provided by Android P.
+///  * [CupertinoPageTransitionsBuilder], which defines a horizontal page
+///    transition that matches native iOS page transitions.
+abstract class PageTransitionsBuilder {
+  /// Abstract const constructor. This constructor enables subclasses to provide
+  /// const constructors so that they can be used in const expressions.
+  const PageTransitionsBuilder();
+
+  /// Wraps the child with one or more transition widgets which define how [route]
+  /// arrives on and leaves the screen.
+  ///
+  /// The [MaterialPageRoute.buildTransitions] method looks up the current
+  /// current [PageTransitionsTheme] with `Theme.of(context).pageTransitionsTheme`
+  /// and delegates to this method with a [PageTransitionsBuilder] based
+  /// on the theme's [ThemeData.platform].
+  Widget buildTransitions<T>(
+    PageRoute<T> route,
+    BuildContext context,
+    Animation<double> animation,
+    Animation<double> secondaryAnimation,
+    Widget child,
+  );
+}
+
+/// Used by [PageTransitionsTheme] to define a default [MaterialPageRoute] page
+/// transition animation.
+///
+/// The default animation fades the new page in while translating it upwards,
+/// starting from about 25% below the top of the screen.
+///
+/// See also:
+///
+///  * [OpenUpwardsPageTransitionsBuilder], which defines a page transition
+///    that's similar to the one provided by Android P.
+///  * [CupertinoPageTransitionsBuilder], which defines a horizontal page
+///    transition that matches native iOS page transitions.
+class FadeUpwardsPageTransitionsBuilder extends PageTransitionsBuilder {
+  /// Construct a [FadeUpwardsPageTransitionsBuilder].
+  const FadeUpwardsPageTransitionsBuilder();
+
+  @override
+  Widget buildTransitions<T>(
+    PageRoute<T> route,
+    BuildContext context,
+    Animation<double> animation,
+    Animation<double> secondaryAnimation,
+    Widget child,
+  ) {
+    return _FadeUpwardsPageTransition(routeAnimation: animation, child: child);
+  }
+}
+
+/// Used by [PageTransitionsTheme] to define a vertical [MaterialPageRoute] page
+/// transition animation that looks like the default page transition
+/// used on Android P.
+///
+/// See also:
+///  * [FadeUpwardsPageTransitionsBuilder], which defines a default page transition.
+///  * [CupertinoPageTransitionsBuilder], which defines a horizontal page
+///    transition that matches native iOS page transitions.
+class OpenUpwardsPageTransitionsBuilder extends PageTransitionsBuilder {
```

Comment:
Open still doesn't feel descriptive. I'm sure there are prior-art from things like PowerPoint or Adobe Premier that we can re-use.

Reply:
So PowerPoint calls it 'Wipe' or 'Cover' and Apple Keynote calls it 'Reveal'. I think any of them are an improvement on 'Open'

Reply:
I think they're all OK, and I'm happy with "open".